src/Uri.php

```diff
@@ -362,7 +362,7 @@ public function toString(bool $encode = false): string
         }
 
         if (! is_null($this->fragment)) {
-            $url .= "$this->fragment";
+            $url .= "#$this->fragment";
```

Comment:
I would check if the fragment already contains `#` before forcing it 👍

Reply:
It does have the '#' by default as part of the `addFragment` method: https://github.com/JustSteveKing/uri-builder/blob/main/src/Uri.php#L305